torch/_inductor/codegen/subgraph.py

```diff
@@ -405,14 +430,17 @@ def _infer_custom_op_layout(
 
         with V.fake_mode:
             example_inputs = []
-            for inp in input_nodes:
-                raw_shape = inp.get_size()
-                concrete_shape = V.graph.sizevars.size_hints(
-                    raw_shape, fallback=config.unbacked_symint_fallback
-                )
-                fake_tensor = torch.empty(
-                    concrete_shape, dtype=inp.get_dtype(), device=inp.get_device()
-                )
+            for i, inp in enumerate(input_nodes):
+                if input_gen_fns and i in input_gen_fns:
+                    fake_tensor = input_gen_fns[i](inp)
+                else:
+                    raw_shape = inp.get_size()
+                    concrete_shape = V.graph.sizevars.size_hints(
+                        raw_shape, fallback=config.unbacked_symint_fallback
+                    )
+                    fake_tensor = torch.empty(
```

Comment:
Can we fix the stride ? some custom ops, e.g. certain gemms actually do require a specific stride.